src/persistentmodel/job.cpp

```diff
@@ -57,8 +58,10 @@ void Job::setUrls(const QList<QUrl> &urls)
 
 bool Job::validateUrls()
 {
+    // Check that _urls is not empty.
```

Comment:
I've been trained to read code by looking at the comments first, and to me, the name `validateUrls` does not imply that `_url` should not be empty.  That comment makes it explicit that this was a deliberate choice.

Basically, if I'm skimming code and I need to read something more than once, I think it's worth adding a comment.  This might be because:
- I didn't understand the code at first glance.
- the comment breaks up a long function (such as the `// Prepare query`, `// Get database instance`, `// Fill in missing values`, `// Run query` comments: it's much easier to find where I am within the function if I can look at 4 comments, instead of 20 lines of code!).
- I didn't understand why the code was doing what it was, but I didn't want to get bogged down in the details.  "Commenting what something does at the moment" seems like a good first step towards investigating / discussing it in the future.  (and yes, this was one case where I *do* want to investigate how `validateUrls` is being used, and quite possibly suggest a name change)

For reference, using the https://github.com/AlDanial/cloc tool to count lines of comments, tarsnap-gui has 2.3% lines of comments (after running `make distclean` to remove the `moc_*.cpp` files):

```
$ ~/Downloads/cloc/cloc --by-percent 'cmb' `find ~/src/my-tarsnap-gui -name "*.cpp"`
Language                     files        blank %      comment %           code
-------------------------------------------------------------------------------
C++                             34           9.16           2.33           7479
```

whereas tarsnap has 22%:
```
$ ~/Downloads/cloc/cloc --by-percent 'cmb' `find ~/src/tarsnap -name "*.c"`
...

Language                     files        blank %      comment %           code
-------------------------------------------------------------------------------
C                              175          12.42          21.95          42106
```

I don't think we need to add 10x the amount of comments in order to "catch up" to the CLI app, but on the other hand I don't think we need to avoid adding a comment if somebody thinks it would be useful.